### PR TITLE
Api 18661 add update bnft claim

### PR DESCRIPTION
**Pull request by acovrig**
## Description of change

Add `updateBnftClaim` endpoint to gem

## Original issue(s)
[API-18661](https://vajira.max.gov/browse/API-18661)

## Things to know about this PR

See comment about `deep_transform_keys`, I don't like my solution, but not sure how else to do it.

### CODE REVIEW

File: lib/bgs/services/benefit_claims.rb [hunk 1]
```diff
@@ -30,5 +30,22 @@ def find_bnft_claim(claim_id:)
...
+      # This works in rails, but not pure ruby
+      # claim.deep_transform_keys! { |key| key.to_s.camelize(:lower).to_sym }
+      # This isn't as pretty, but works in pure ruby
+      claim.transform_keys! do |key|
+        str = key.to_s.downcase.split('_').collect(&:capitalize).join
+        str[0] = str[0].downcase
+        str.to_sym
+      end
+
+      response = request(:update_bnft_claim, { bnftClaimDTO: claim }, claim[:bnftClaimId])
```
Comment (FonzMP) [inline lib/bgs/services/benefit_claims.rb:47]: doesn't the find give us a response with 
```
{:bnft_claim_dto=>{:bnft_claim_id=>
```
looking at the PR for vets-api, it looks like the claim comes back with the above.
if that's the case we wouldn't necessarily need `bnftClaimDTO` in 47, but we could actually keep that and just do `claim[:bnft_claim_dto].transform_keys!` in 41 and pass that to 47 as the claim so we don't have to do some weird double nesting with `transform_keys!`

File: spec/bgs/services/benefit_claims_spec.rb [hunk 2]
```diff
@@ -48,4 +48,18 @@
       expect(response[:bnft_claim_dto][:bnft_claim_id]).to eq(test_claim_id)
     end
   end
+
+  it 'updates benefit claim' do
+    VCR.use_cassette('benefit_claims/update_bnft_claim') do
+      claim = service.benefit_claims.find_bnft_claim(claim_id: test_claim_id)[:bnft_claim_dto]
```
Comment (FonzMP) [inline spec/bgs/services/benefit_claims_spec.rb:54]: if the above is true, we wouldn't need the last part of this ( `[:bnft_claim_dto]` )

**Comment by acovrig** [comment]
Comment from @FonzMP:

> my thought was to assume (I hate assuming, but ? ) that we'd always have the `bnft_claim_dto` on the claim itself... then change the transform to be
`camel_this_claim = claim[:bnft_claim_dto].transform_keys! do |key|`
so we're just transforming and returning the keys inside that... then we don't have to do any trickery to transform the `bnft_claim_dto` and at the same time, we convert the keys within that to camelCase.
 once we have the aforementioned, we could pass that.... so we could do something like
`response = request(:update_bnft_claim, { bnftClaimDto: camel_this_claim }, claim[:bnftClaimId])`
also, we're further out of the claim to get the ID, so `claim[:bnftClaimId]` will be nil in the current state. The only thing we'll have access to at the claim level is `bnftClaimDto` currently.

True, however, I like the deep transform - what if the data that comes back ends up with a nested field (if BGS adds a field) then that field would fail to update in the future (since the sub-fields wouldn't get camelized).

**Comment by FonzMP** [comment]
> True, however, I like the deep transform - what if the data that comes back ends up with a nested field (if BGS adds a field) then that field would fail to update in the future (since the sub-fields wouldn't get camelized).

so your thought is to create a method that would dive inside each "layer" and camelCase the keys? @acovrig

**Comment by acovrig** [comment]
> > True, however, I like the deep transform - what if the data that comes back ends up with a nested field (if BGS adds a field) then that field would fail to update in the future (since the sub-fields wouldn't get camelized).
> 
> so your thought is to create a method that would dive inside each "layer" and camelCase the keys? @acovrig

yes

File: spec/fixtures/vcr_cassettes/benefit_claims/update_bnft_claim.yml [hunk 3]
```diff
@@ -0,0 +1,919 @@
+---
```
Comment (FonzMP) [inline spec/fixtures/vcr_cassettes/benefit_claims/update_bnft_claim.yml:1]: This is the cassette for the `find` and `update`. I'd vote we rerecord or remove the `find` (first 2 requests).

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**